rclcpp/include/rclcpp/strategies/static_memory_strategy.hpp

```diff
@@ -27,40 +27,59 @@ namespace memory_strategies
 
 namespace static_memory_strategy
 {
-  struct ObjectPoolBounds
-  {
-  public:
-    size_t max_subscriptions_;
-    size_t max_services_;
-    size_t max_clients_;
-    size_t max_executables_;
-    size_t pool_size_;
-
-    ObjectPoolBounds(size_t subs = 10, size_t services = 10, size_t clients = 10,
-      size_t executables = 1, size_t pool = 1024)
-    : max_subscriptions_(subs), max_services_(services), max_clients_(clients), max_executables_(
-        executables), pool_size_(pool)
-    {}
-  };
+struct ObjectPoolBounds
+{
+public:
+  size_t max_subscriptions_;
+  size_t max_services_;
+  size_t max_clients_;
+  size_t max_executables_;
+  size_t max_guard_conditions_;
+  size_t pool_size_;
+
+  ObjectPoolBounds(size_t subs = 10, size_t services = 10, size_t clients = 10,
+    size_t executables = 1, size_t guard_conditions = 2, size_t pool = 1024)
+  : max_subscriptions_(subs), max_services_(services), max_clients_(clients),
+    max_executables_(executables), max_guard_conditions_(guard_conditions), pool_size_(pool)
+  {}
+};
 
 
 class StaticMemoryStrategy : public memory_strategy::MemoryStrategy
 {
 public:
   StaticMemoryStrategy(ObjectPoolBounds bounds = ObjectPoolBounds())
-  : bounds_(bounds)
+  : bounds_(bounds), memory_pool_(nullptr), subscription_pool_(nullptr),
+    service_pool_(nullptr), guard_condition_pool_(nullptr), executable_pool_(nullptr)
   {
-    memory_pool_ = static_cast<void **>(malloc(bounds_.pool_size_));
-    subscription_pool_ = static_cast<void **>(malloc(bounds_.max_subscriptions_));
-    service_pool_ = static_cast<void **>(malloc(bounds_.max_services_));
-    client_pool_ = static_cast<void **>(malloc(bounds_.max_clients_));
-    executable_pool_ = static_cast<executor::AnyExecutable *>(malloc(bounds_.max_executables_));
-
-    memset(memory_pool_, 0, bounds_.pool_size_);
-    memset(subscription_pool_, 0, bounds_.max_subscriptions_);
-    memset(service_pool_, 0, bounds_.max_services_);
-    memset(client_pool_, 0, bounds_.max_clients_);
-    memset(executable_pool_, 0, bounds_.max_executables_);
+    if (bounds_.pool_size_) {
+      memory_pool_ = new void *[bounds_.pool_size_];
```

Comment:
yes, they should. I've restored the `memset` calls so that the memory that the `pool` objects is initialized to 0.